subscription_test.go

```diff
@@ -369,3 +370,59 @@ func TestEventStreamEventsReceived(t *testing.T) {
 		assert.Fail(t, "channel was not closed")
 	}
 }
+
+func TestRegisterSSESubscriptionMarksMemberDownAndRetries(t *testing.T) {
+	clientCfg := NewDefaultConfig()
+	// Use non-existent address as first cluster member
+	clientCfg.URL = "http://127.0.0.1:11111"
+	clientCfg.EventsTransport = EventsTransportSSE
+	config := configContainer{client: &clientCfg}
+
+	endpoint := newFakeMarathonEndpoint(t, &config)
+	defer endpoint.Close()
+
+	client := endpoint.Client.(*marathonClient)
+	// Add real server as member to the cluster
+	client.hosts.members = append(client.hosts.members, &member{endpoint: endpoint.Server.httpSrv.URL})
+
+	assert.Equal(t, 2, len(client.hosts.activeMembers()), "There should be 2 active members in the cluster")
+
+	// The connection should work as it will use the second member of the cluster
+	_, err := endpoint.Client.AddEventsListener(EventIDApplications)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(client.hosts.activeMembers()), "There should be 1 active member in the cluster")
+}
+
+func TestRegisterSEESubscriptionReconnectsStreamOnError(t *testing.T) {
+	clientCfg := NewDefaultConfig()
+	clientCfg.EventsTransport = EventsTransportSSE
+	config := configContainer{client: &clientCfg}
+
+	endpoint1 := newFakeMarathonEndpoint(t, &config)
+	endpoint2 := newFakeMarathonEndpoint(t, &config)
+	defer endpoint2.Close()
+
+	client1 := endpoint1.Client.(*marathonClient)
+	// Add the second server to the cluster members
+	client1.hosts.members = append(client1.hosts.members, &member{endpoint: endpoint2.Server.httpSrv.URL})
+
+	events, err := endpoint1.Client.AddEventsListener(EventIDApplications)
+	require.NoError(t, err)
+
+	// This should make the SSE subscription to fail and reconnect to another cluster member
+	endpoint1.Close()
+
+	// Let a bit time so that subscription can reconnect
+	time.Sleep(10 * time.Millisecond)
```

Comment:
As mentioned offline, would be great to have a more deterministic approach here. That is, something that tells us reliably when the subscription has been reconnected.